lib/apartment/elevators/subdomain.rb

```diff
@@ -38,13 +43,23 @@ def subdomain(host)
       end
 
       def subdomains(host)
-        return [] unless named_host?(host)
+        host_valid?(host) ? parse_host(host) : []
+      end
+
+      def host_valid?(host)
+        not ip_host?(host) and domain_valid?(host)
+      end
+
+      def ip_host?(host)
+        !/^[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}$/.match(host).nil?
+      end
 
-        host.split('.')[0..-(Apartment.tld_length + 2)]
+      def domain_valid?(host)
+        PublicSuffix::VERSION[0] == '1' ? PublicSuffix.valid?(host) : PublicSuffix.valid?(host, ignore_private: true)
```

Comment:
I think if we will add a new dependency, that be the newest version, not the old one.